rocky/katalogus/client.py

```diff
@@ -121,181 +144,200 @@ def __init__(self):
-            response.raise_for_status()
-        except httpx.HTTPStatusError as error:
-            if error.response.status_code == codes.CREATED:
+            if response.status_code == codes.CREATED:
                 logger.info("Plugin %s created", plugin.name)
             else:
                 logger.info("Plugin %s could not be created", plugin.name)
-            error_message = json.loads(error.response.text).get("detail")
-            if error.response.status_code == codes.BAD_REQUEST and "Duplicate plugin" in error_message:
-                raise DuplicatePluginError(error_message)
-            else:
-                raise error
+        except KATalogusHTTPStatusError:
+            logger.info("Plugin %s could not be created", plugin.name)
+            raise
 
-    def edit_plugin(self, plugin: Plugin) -> None:
+    def edit_plugin(self, organization_code: str, plugin: Plugin) -> None:
         try:
             response = self.session.patch(
-                f"{self.organization_uri}/boefjes/{plugin.id}", content=plugin.model_dump_json(exclude_none=True)
+                f"/v1/organisations/{organization_code}/boefjes/{plugin.id}",
+                content=plugin.model_dump_json(exclude_none=True),
             )
-            response.raise_for_status()
-        except httpx.HTTPStatusError as error:
-            if error.response.status_code == codes.CREATED:
+            if response.status_code == codes.CREATED:
                 logger.info("Plugin %s updated", plugin.name)
             else:
                 logger.info("Plugin %s could not be updated", plugin.name)
-            if error.response.status_code == codes.BAD_REQUEST and "duplicate key" in error.response.text:
-                raise DuplicatePluginError("Duplicate plugin name")
-            if error.response.status_code in [codes.FORBIDDEN, codes.NOT_FOUND]:
-                raise KATalogusNotAllowedError
-            else:
-                raise error
+        except KATalogusHTTPStatusError:
+            logger.info("Plugin %s could not be updated", plugin.name)
+            raise
+
+    def _patch_plugin_state(self, organization_code: str, plugin_id: str, enabled: bool) -> None:
+        logger.info("Toggle plugin state", plugin_id=plugin_id, enabled=enabled)
+        plugin_id = quote(plugin_id)
+
+        self.session.patch(f"/v1/organisations/{organization_code}/plugins/{plugin_id}", json={"enabled": enabled})
+
+
+class KATalogus:
+    """
+    An adapter between the full KATalogusClient and the organization-specific context of most views. This restricts
+    the set of available methods on the KATalogusClient and simplifies the interface by "currying" the organization
+    into the relevant methods as an argument. We should use this class in the views to avoid making calls exposing
+    information from other organizations users are not allowed to see.
+    """
+
+    def __init__(self, katalogus_client: KATalogusClient, organization_code: str):
```

Comment:
Why not subclass from KATalogusClient, and add the functionality that is specific to the organization here?

Reply:
Can we not pass the member as argument, as a member has many orgs from the one too many relationship. Then we can list all orgs that belongs to member, then we can check if a specific org is in this list.

Reply:
Because that would still expose the non-organization-specific methods to users of the class. It also implies that this class has the same interface of the parent, but we actually change the interface because we remove the organization as an argument. These were the main two considerations for not extending the parent class and creating a bit of "duplication" although not literal duplication.